src/commands/crashlytics-mappingfile-upload.ts

```diff
@@ -0,0 +1,63 @@
+import { Command } from "../command";
+import { FirebaseError } from "../error";
+import * as utils from "../utils";
+
+import { fetchBuildtoolsJar, runBuildtoolsCommand } from "../crashlytics/buildToolsJarHelper";
+
+interface CommandOptions {
+  app: string | null;
+  mappingFile: string | null;
+  resourceFile: string | null;
+  debug: boolean | null;
+}
+
+interface JarOptions {
+  app: string;
+  mappingFilePath: string;
+  resourceFilePath: string;
+}
+
+export default new Command("crashlytics:mappingfile:upload <mappingFile>")
+  .description("Upload a mapping file to deobfuscate stack traces.")
+  .option("--app <appID>", "The app id of your Firebase app")
+  .option(
+    "--resource-file <resourceFile>",
+    "Path to the Android resource XML file that includes the mapping file id."
+  )
+  .option("--debug", "print debug output and logging from the underlying uploader tool")
+  .action(async (mappingFile: string, options: CommandOptions) => {
+    const app = getGoogleAppID(options) || "";
+    const debug = !!options.debug;
+    if (!mappingFile) {
+      throw new FirebaseError("set <mappingFile> to a valid mapping file path");
```

Comment:
Could we give more detail here about the required format of the path?

Reply:
I added examples of the path to all the errors in the next commit.